dowhy/causal_estimator.py

```diff
@@ -679,6 +685,148 @@ def signif_results_tostr(self, signif_results):
+            # This is done as all dowhy estimators have two parts and external ones have two or more parts
+            if num_components > 2:
+                estimator_package = estimator_name.split(".")[0]
+                if estimator_package == "dowhy":  # For updated dowhy methods
+                    estimator_method = estimator_name.split(".", maxsplit=1)[
+                        1
+                    ]  # discard dowhy from the full package name
+                    causal_estimator_class = causal_estimators.get_class_object(estimator_method + "_estimator")
+                else:
+                    third_party_estimator_package = estimator_package
+                    causal_estimator_class = causal_estimators.get_class_object(
+                        third_party_estimator_package, estimator_name
+                    )
+                    if method_params is None:
+                        method_params = {}
+                    # Define the third-party estimation method to be used
+                    method_params[third_party_estimator_package + "_methodname"] = estimator_name
+            else:  # For older dowhy methods
+                logger.info(estimator_name)
+                # Process the dowhy estimators
+                causal_estimator_class = causal_estimators.get_class_object(estimator_name + "_estimator")
+        if identified_estimand.no_directed_path:
+            logger.warning("No directed path from {0} to {1}.".format(treatment, outcome))
+            return CausalEstimate(
+                0, identified_estimand, None, control_value=control_value, treatment_value=treatment_value
+            )
+        # Check if estimator's target estimand is identified
+        elif identified_estimand.estimands[identifier_name] is None:
+            logger.error("No valid identified estimand available.")
+            return CausalEstimate(None, None, None, control_value=control_value, treatment_value=treatment_value)
+        else:
+
+            if method_params is not None and (num_components <= 2 or estimator_package == "dowhy"):
+                extra_args = method_params.get("init_params", {})
+            else:
+                extra_args = {}
+            if method_params is None:
+                method_params = {}
+            causal_estimator = causal_estimator_class(
+                data,
+                identified_estimand,
+                treatment,
+                outcome,  # names of treatment and outcome
+                control_value=control_value,
+                treatment_value=treatment_value,
+                test_significance=test_significance,
+                evaluate_effect_strength=evaluate_effect_strength,
+                confidence_intervals=confidence_intervals,
+                target_units=target_units,
+                effect_modifiers=effect_modifiers,
+                **method_params,
+                **extra_args,
+            )
+    else:
```

Comment:
Right... that was confusing for me, I don't like the fit_estimator flag as that seems to be a functionality of the estimator instead of this function. I'll change the input parameter to be an object so that we can keep that functionality for this PR and modify this method in future PRs to add the `fit()` call

Reply:
I moved all the code to initialize the estimator into the causal_model (for compatibility) and made the function accept the object as parameter, so that it works the way you suggest